src/ModelContextProtocol/Protocol/Types/Resources/TextResourceContents.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Text.Json.Serialization;
+
+namespace ModelContextProtocol.Protocol.Types;
+
+/// <summary>
+/// Text contents of a resource.
+/// <see href="https://github.com/modelcontextprotocol/specification/blob/main/schema/2024-11-05/schema.json">See the schema for details</see>
+/// </summary>
+public class TextResourceContents : ResourceContents
+{
+    /// <summary>
+    /// The text content of the resource.
```

Comment:
```suggestion
    /// The text of the item. This must only be set if the item can actually be represented as text (not binary data).
```
(from spec description)